src/docdb/nodes.ts

```diff
@@ -91,34 +98,39 @@ export class DocDBCollectionNode implements INode {
 		return this.db.getDbLink() + '/colls/' + this.id;
 	}
 
-	async getDocuments(): Promise<any> {
-		const dbLink: string = this.db.getDbLink();
-		const client = new DocumentClient(this.db.getEndpoint(), { masterKey: this.db.getPrimaryMasterKey() });
-		const collSelfLink = this.getCollLink();
-		const docs = await this.readOneCollection(collSelfLink, client);
-		return await docs;
+	clearCache(): void {
+		this._children = null;
+		this._hasFetched = false;
+		this._loadMoreNode = null;
 	}
 
 	async getChildren(): Promise<INode[]> {
-		const collLink: string = this.getCollLink();
-		const parentNode = this;
-		const client = new DocumentClient(this.db.getEndpoint(), { masterKey: this.db.getPrimaryMasterKey() });
-		let documents = await this.listDocuments(collLink, client);
-		return documents.map(document => new DocDBDocumentNode(document.id, parentNode, document));
-	}
-
-	async listDocuments(collSelfLink, client): Promise<any> {
-		let documents = await client.readDocuments(collSelfLink);
-		return await new Promise<any[]>((resolve, reject) => {
-			documents.toArray((err, cols: Array<Object>) => err ? reject(err) : resolve(cols));
-		});
-	}
-
-	async readOneCollection(selfLink, client): Promise<any> {
-		let documents = await client.readDocuments(selfLink, { maxItemCount: 20 });
-		return await new Promise<any[]>((resolve, reject) => {
-			documents.toArray((err, docs: Array<Object>) => err ? reject(err) : resolve(docs));
-		});
+		if (!this._hasFetched) {
+			const collLink: string = this.getCollLink();
+			const client = new DocumentClient(this.db.getEndpoint(), { masterKey: this.db.getPrimaryMasterKey() });
+			let docIterator = await client.readDocuments(collLink);
+			const elements = await LoadMoreNode.loadMore(docIterator, this._batchSize);
+			const documents = elements.results;
+			if (elements.hasMore) {
+				this._loadMoreNode = new LoadMoreNode(docIterator, this);
+			}
+			else {
+				this._loadMoreNode = null;
+			}
+			this._children = this._children.concat(documents.map(document => new DocDBDocumentNode(document.id, this, document)));
+			this._hasFetched = true;
+		}
+		return this._children.concat([this._loadMoreNode]);
+
+	}
+
+	async addMoreChildren(): Promise<void> {
+		const elements = await LoadMoreNode.loadMore(this._loadMoreNode.iterator, this._batchSize);
```

Comment:
It doesn't make sense to call a static method on LoadMoreNode with state from an instance of a LoadMoreNode. Either store the state somewhere else or make the method non-static and use the state directly (I prefer storing the iterator in DocDBCollectionNode)